pkg/util/util.go

```diff
@@ -0,0 +1,27 @@
+package util
+
+import (
+	"fmt"
+	"strings"
+)
+
+// ParseLabels takes a comma-separated string of key-value pairs and returns a map of labels.
+func ParseLabels(labelStr string) (map[string]string, error) {
+	labels := make(map[string]string)
+
+	if len(labelStr) == 0 {
+		return labels, fmt.Errorf("empty string provided")
+	}
+	pairs := strings.Split(labelStr, ",")
+
+	for _, pair := range pairs {
+		keyValue := strings.SplitN(pair, "=", 2)
+		if len(keyValue) != 2 {
```

Comment:
If you use SplitN then keyValue will only be != 2 if there are no '=' in the input. In this case I think it would be safer to use Split. This will allow "foo=bar=err" to return {"foo", "bar", "err"} and cause an error for length 3. As it stands SplitN will return {"foo", "bar=err"} which has length 2.